sbi/utils/posterior_ensemble.py

```diff
@@ -0,0 +1,196 @@
+            ensemble.
+        Returns:
+            `NeuralPosteriorEnsemble` that will use a default `x` when not explicitly
+            passed.
+        """
+        for posterior in self.posteriors:
+            posterior.set_default_x(x)
+        return self
+
+    def map(self, individually: bool = False, **kwargs) -> Tensor:
+        r"""Returns the average maximum-a-posteriori estimate (MAP).
+
+        Computes MAP estimate for each posterior in the ensemble individually and
+        averages them. All args and kwargs are passed directly through to
+        `posterior.map()`.
+
+        Since `posterior.map()` is computed one by one, the subroutines of each MAP
+        estimate can be interrupted individually (Ctrl-C), when the user sees that the
+        log-probability converges. The best estimate will be saved in `self.posteriors
+        [idx].map_` and added to the average MAP estimate.
+
+        For more details of how the MAP estimate is obtained see `.map()` docstring of
+        self.posteriors[idx].
+
+        Args:
+            individually: If true, returns log weights and MAPs individually.
+
+        Kwargs:
+            x: Observed data at which to evaluate the MAP.
+            num_iter: Number of optimization steps that the algorithm takes
+                to find the MAP.
+            learning_rate: Learning rate of the optimizer.
+            init_method: How to select the starting parameters for the optimization. If
+                it is a string, it can be either [`posterior`, `prior`], which samples
+                the respective distribution `num_init_samples` times. If it is a
+                tensor, the tensor will be used as init locations.
+            num_init_samples: Draw this number of samples from the posterior and
+                evaluate the log-probability of all of them.
+            num_to_optimize: From the drawn `num_init_samples`, use the
+                `num_to_optimize` with highest log-probability as the initial points
+                for the optimization.
+            save_best_every: The best log-probability is computed, saved in the
+                `map`-attribute, and printed every `save_best_every`-th iteration.
+                Computing the best log-probability creates a significant overhead
+                (thus, the default is `10`.)
+            show_progress_bars: Whether or not to show a progressbar for sampling from
+                the posterior.
+            log_prob_kwargs: Will be empty for SNLE and SNRE. Will contain
+                {'norm_posterior': True} for SNPE.
+
+        Returns:
+            The average MAP estimate.
+        """
+        maps = []
+        log_weights = torch.log(self._weights).reshape(-1, 1)
+
+        for posterior in self.posteriors:
+            maps.append(posterior.map(**kwargs))
+        maps = torch.stack(maps)
+
+        if individually:
+            return log_weights, maps
+        else:
```

Comment:
Hmmm it would be great to compute the **actual** map of the ensemble. I.e. run [`gradient_ascent`](https://github.com/mackelab/sbi/blob/5e74f757b182753714da20b3714b99bef3000644/sbi/inference/posteriors/base_posterior.py#L171-L181) on the mixture

Reply:
That is a good suggestion, I think that won't be too difficult. :)